Source/WebCore/dom/Document.cpp

```diff
@@ -11135,20 +11140,27 @@ void Document::securityOriginDidChange()
 
 #if ENABLE(CONTENT_EXTENSIONS)
 
-ResourceMonitor* Document::resourceMonitor()
+ResourceMonitor* Document::resourceMonitorIfExists()
 {
     return m_resourceMonitor.get();
 }
 
-void Document::setResourceMonitor(RefPtr<ResourceMonitor>&& resourceMonitor)
+ResourceMonitor& Document::resourceMonitor()
+{
+    if (!m_resourceMonitor)
+        m_resourceMonitor = ResourceMonitor::create(*frame());
```

Comment:
Should we asserting that this is not the top document?

Reply:
Added